include/list.h

```diff
@@ -326,7 +327,9 @@ typedef struct xLIST
         }                                                                        \
                                                                                  \
         ( pxItemToRemove )->pxContainer = NULL;                                  \
-        ( pxList->uxNumberOfItems )--;                                           \
+        uxNumberOfItems = ( pxList->uxNumberOfItems );                           \
```

Comment:
I would add a comment explaining why this temp variable is being used.
It might be tempting in the future to revert this change if it is not understood why this is being done.

Reply:
Agreed, will do that now